packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,731 @@
+2. `shape-list`
+    - A component that visualizes the shapes that we have added
+
+And finally we will update `shape-dials` to disable the buttons when they are not applicable. The application should look something like this:
+
+<img
+    src="/images/articles/redux-reactive-controllers/application.webp"
+    loading="lazy"
+    width="550px"
+    title="Sketch of the redux application"
+    alt="A vector box diagram of the entire redux application. There are 3 sections. The first is the shape-dials component – it has 3 rows. The first row is a minus button, the label 'squares', followed by a plus button. The second row is a minus button, the label 'circles', followed by a plus button. The final row is a long 'reset' button. This component is surrounded by a red dashed-square labeled '<shape-dials>'. The next section to the right of the shape-dials section is labeled by a red label that says '<shape-count>'. It is a black box with 3 rows of text. The first row is '2 circles'. The second row is '2 squares', and the third row is '4 total'. Below the shape-dials and shape-count components is a long rectangle labeled '<shape-list>'. This is a long, horizonal, black rectangle and inside of it is a red square followed by a blue circle, blue circle, and another red square.">
+
+### shape-count
+
+`shape-count` should be a component that only subscribes and reads from the store. Let us create a custom element and render the table:
+
+```ts
+import { LitElement, html} from 'lit';
+import { customElement} from 'lit/decorators.js';
+
+@customElement('shape-count')
+export class ShapeCount extends LitElement {
+  render() {
+    return html`
+      <table>
+        <tr>
+          <td>${0}</td>
+          <td>circles</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>squares</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>total</td>
+        </tr>
+      </table>
+    `;
+  }
+}
+```
+
+Next we want to render actual counts rather than just `0`. In this case we will need all values from the state:
+
+- `state.circles`
+- `state.squares`
+- `state.shapeList`
+
+To achieve this, we will need a broad Redux selector that selects the entire state:
+
+```ts
+...
+import { SelectorController } from './selector-controller.js';
+import { store } from './store.js';
+
+@customElement('shape-count')
+export class ShapeCount extends LitElement {
+  private sc = new SelectorController(this, store, (state) => state);
+
+  render() {
+    const {circles, squares, shapeList} = this.sc.selected;
+
+    return html`
+      <table>
+        <tr>
+          <td>${circles}</td>
+          <td>circles</td>
+        </tr>
+        <tr>
+          <td>${squares}</td>
+          <td>squares</td>
+        </tr>
+        <tr>
+          <td>${shapeList.length}</td>
+          <td>total</td>
+        </tr>
+      </table>
+    `;
+  }
+}
+```
+
+Pressing the buttons in `shape-dials` should now update the count!
```

Comment:
It's a bit hard to follow this part since we haven't actually seen `shape-dials`.. maybe that's okay for now.

Reply:
Does the image in the previous section nor the image in the Actions section not help?

Reply:
I think my uneasiness with this sentence is more that this describes what clicking the buttons in `shape-dials` does which isn't the what the previous code is about. It's about subscribing to the store and updating UI on state change.

Maybe it can be something like
```suggestion
The counts are now pulled from the state in Redux and any updates to the state will update the component!
```

Reply:
ah this makes sense